TshepoMagagula/To-Do-List
# Add To-Do List Functionality with CRUD Operations

This pull request introduces a fully functional to-do list web application, allowing users to create, view, update, and delete to-do items. The app provides a clean and intuitive interface for managing daily tasks and includes essential CRUD (Create, Read, Update, Delete) functionality.

Features Added:

Create Tasks - Users can add new to-do items by entering a task name and pressing the "Add" button.
Read Tasks - All tasks are displayed in a list format with their status (completed or pending).
Update Tasks - Tasks can be marked as completed.
Delete Tasks - Users can remove tasks from the list.
Persistent Storage - Tasks are saved using an array on the server and are rendered on the page loader, so they remain available after page refresh.
Responsive Design - The application layout adjusts for both mobile and desktop devices.

Tech Stack:

Frontend: HTML, CSS, JavaScript (vanilla)
Storage: LocalStorage for persisting tasks across sessions.

How to Test:

Clone the repository and switch to this branch.
Open index.html in a browser.
Test all CRUD operations (create, mark complete, and delete) and ensure tasks persist after refreshing the page.
Confirm responsive design by testing on different screen sizes.
Screenshots: Include screenshots of the app's main views for reference.

Next Steps:

Add categories/tags for task organization.
Implement search/filter functionality for large task lists.
Notes: Feel free to provide feedback on code organization and UI/UX enhancements.
